linera-core/src/client/chain_client_state.rs

```diff
@@ -50,7 +53,7 @@ impl ChainClientState {
         timestamp: Timestamp,
         next_block_height: BlockHeight,
         pending_block: Option<ProposedBlock>,
-        pending_blobs: BTreeMap<BlobId, Blob>,
+        pending_blobs: impl IntoIterator<Item = Blob>,
```

Comment:
In theory, this iterator can be non-empty for `pending_block = None`. What do you think about asserting that they're consistent?